cas-server-core/src/main/java/org/jasig/cas/authentication/principal/DefaultPrincipalAttributesRepository.java

```diff
@@ -30,11 +32,19 @@
  * @author Misagh Moayyed
  * @since 4.1
  */
-public final class DefaultPrincipalAttributesRepository implements PrincipalAttributesRepository {
+public final class DefaultPrincipalAttributesRepository extends AbstractPrincipalAttributesRepository {
     private static final long serialVersionUID = -4535358847021241725L;
 
     @Override
-    public Map<String, Object> getAttributes(final Principal p) {
+    protected void addPrincipalAttributesIntoCache(final String id, final Map<String, Object> attributes) {
```

Comment:
It feels a bit strange to find "cache"  methods in the default implementation which has no cache.